app/javascript/redux/greetingSlice.js

```diff
@@ -0,0 +1,64 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+export const fetchGreeting = createAsyncThunk(
+  "greetings/fetchGreeting",
+  async () => {
+    const response = await axios.get("http://127.0.0.1:3000/api/v1/greeting");
+    const formattedRes = await response.data;
+    return formattedRes;
+  }
+);
+
+export const greetingSlice = createSlice({
+  name: "greetings",
+  initialState: {
+    greeting: {},
+    status: null,
+  },
+
+  extraReducers: {
+    [fetchGreeting.pending]: (state) => {
+      state.status = "Loading";
+    },
+    [fetchGreeting.fulfilled]: (state, action) => {
+      state.greeting = action.payload;
+      state.status = "Success";
+    },
+    [fetchGreeting.rejected]: (state) => {
+      state.status = "Failed";
+    },
+  },
+});
+
+export default greetingSlice.reducer;
+
+// const FETCH = "greeting/fetchRandomGreeting";
+
+// const initialState = {};
+
+// const greetingReducer = (state = initialState, action) => {
+//   switch (action.type) {
+//     case FETCH:
+//       return { ...state, ...action.payload };
+//   }
+// };
+
+// export const getGreeting = (data) => {
+//   return {
+//     type: FETCH,
+//     payload: data,
+//   };
+// };
+
+// export const fetchGreeting = () => async (dispatch) => {
+//   try {
+//     await fetch("http://127.0.0.1:3000/api/v1/greeting").then((response) =>
+//       dispatch(getGreeting(response.json()))
+//     );
+//   } catch (err) {
+//     throw new Error(err);
+//   }
+// };
+
+// export default greetingReducer;
```

Comment:
- [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

Reply:
Thank you!